### PR TITLE
Add date picker

### PR DESCRIPTION
**Pull request by binaryberry**
Users can now pick dates from calendar. Added jQuery UI date picker, jasmine for Javascript testing. Parsing date parameters through to support-api.

cc @boffbowsh @fofr @benilovj 

## Selecting a date range
screen shot 2015-05-20 at 09 57 34
screen shot 2015-05-20 at 09 57 47

## When no feedback in date range
screen shot 2015-05-21 at 10 40 28

## Open ended date ranges and lenient date formats
screen shot 2015-05-20 at 09 58 04
screen shot 2015-05-20 at 09 58 32
screen shot 2015-05-20 at 09 58 16

## Warnings when filter fails
screen shot 2015-05-21 at 10 42 31

### CODE REVIEW

**Comment by fofr** [comment]
This is working nicely. :+1:

I have some tweaks to suggest around the way the filter/URL is working with partial date ranges and custom date formats – I think user input that doesn't match the specified format, or which provides only partial information should be coerced into the correct format.

The feedback order (reverse-chronological) when filtered feels counterintuitive.

We're also missing an easy way to to remove the filter which I forgot to include.

These are mostly kinks that a non-functional prototype wouldn't catch.

**Comment by benilovj** [comment]
A small comment: the last two commits were made by the vagrant user; it'd be best to amend those with proper author data, and set up git within the VM with your credentials...

**Comment by boffbowsh** [comment]
Thanks for adding screenshots, @binaryberry

File: app/helpers/date_filter_helper.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module DateFilterHelper
```
Comment (benilovj) [inline app/helpers/date_filter_helper.rb]: indentation doesn't look quite right on this module

**Comment by benilovj** [comment]
Apart from minor style remark, :+1:

File: app/helpers/date_filter_helper.rb (hunk 1 above)
Comment (binaryberry) [inline app/helpers/date_filter_helper.rb]: True! Sorted now.

**Comment by benilovj** [comment]
:+1:

File: app/helpers/date_filter_helper.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  def total_responses_header(total_count, from, to)
+    response_total = pluralize(number_with_delimiter(total_count), 'response')
+
+    if from.present? and to.present?
+      "#{response_total} between #{from} and #{to}"
+    elsif from.present?
+      "#{response_total} since #{from}"
+    elsif to.present?
+      "#{response_total} before #{to}"
+    elsif total_count && total_count > 1
```
Comment (tadast) [inline app/helpers/date_filter_helper.rb]: Is there a reason why there are both `and` and `&&` here? :)

Comment (binaryberry) [inline app/helpers/date_filter_helper.rb]: Not really @tadast ! This PR has been merged now but I'll turn the "and" to "&&" in another branch we are working on. Consistency IS good :-)

**Pull request merged**